cmd/thanos/main.go

```diff
@@ -295,21 +296,25 @@ func defaultGRPCServerOpts(logger log.Logger, reg *prometheus.Registry, tracer o
 }
 
 // metricHTTPListenGroup is a run.Group that servers HTTP endpoint with only Prometheus metrics.
-func metricHTTPListenGroup(g *run.Group, logger log.Logger, reg *prometheus.Registry, httpBindAddr string) error {
+func metricHTTPListenGroup(g *run.Group, logger log.Logger, reg *prometheus.Registry, httpBindAddr string, component component.Component) (*prober.Prober, error) {
 	mux := http.NewServeMux()
 	registerMetrics(mux, reg)
 	registerProfile(mux)
+	readinessProber := prober.NewProber(component, logger)
+	readinessProber.RegisterInMux(mux)
 
 	l, err := net.Listen("tcp", httpBindAddr)
 	if err != nil {
-		return errors.Wrap(err, "listen metrics address")
+		return nil, errors.Wrap(err, "listen metrics address")
 	}
 
 	g.Add(func() error {
 		level.Info(logger).Log("msg", "Listening for metrics", "address", httpBindAddr)
+		readinessProber.SetHealthy()
```

Comment:
I think readiness should go here as well?

Reply:
Hmm not sure about this. I'd say when the component is ready should be specific for each one not being set automatically when metrics are exposed by this generic metrics router.